### PR TITLE
option to specify label font color

### PR DESCRIPTION
**Pull request by Batwam**
implement solution for issue #93 to specify label font color

image

image

### CODE REVIEW

**Comment by Batwam** [comment]
updated to include a color choose widget (and use rgb value in the background): 
image

**Comment by Moon-0xff** [comment]
I think this option shouldn't have a default value (or have a *null* value as default), and the code should be executed only if the user has picked something.  
Font colors are dependent on the shell theme, and our default might blend with the background.

**Comment by Batwam** [comment]
I agree with the fact that assuming that white is the default can be dangerous if people have themes. The main challenge I had is that, while it's probably possible to figure it out, I don't know how to find out what that default colour is. This said, I could leave the default setting blank. This will have the effect to set_style('color:""') which seems to set the colour back to the default. 

The Setting page then looks like this with a transparent colour (since the colour isn't recognised) which indicates that the default font isn't being overwritten): 
image

Is that good/clear enough? If not, we'd need to find a way to extract the default colour so it can be applied to the ColorButton when the setting is blank.

**Comment by Batwam** [comment]
ok, I found how to get the font colour using [this post](https://discourse.gnome.org/t/how-to-get-data-from-extension-js-file-to-prefs-js/15095). As far as I can tell, it's not possible to import `Main` in prefs.js (which was also the issue the person posting had) so I followed the same solution and created a setting which is used to store this value. This setting is being populated/refreshed on started and whenever the color  is getting changed. 

The sole reason I'm doing this is so I can set the colorPicker's colour correctly in the preferences window to match the theme's font when the font color is being reset. Otherwise, the previous solution with the checkered rectangle works as well doesn't necessitate the additional code.

**Comment by Batwam** [comment]
I have reverted the previous commit. it was possible to find the font colour but this added too much complexity to the code and potential hedge cases when trying to keep `addColorPicker()` generic for future uses, depending on whether an alternative key exists which needs to be checked or not, effect of the theme changing after the Settings window is being opened, ...

File: schemas/org.gnome.shell.extensions.mpris-label.gschema.xml [hunk 1]
```diff
@@ -30,6 +30,14 @@
...
     </key>
 
+    <key name="font-color" type="s">
+      <default>""</default>
+    </key>
+
+    <key name="font-color-default" type="s">
+      <default>""</default>
+    </key>
+
```
Comment (Moon-0xff) [inline schemas/org.gnome.shell.extensions.mpris-label.gschema.xml:40]: Unused, leftover from [e6ddde9](https://github.com/Moon-0xff/gnome-mpris-label/pull/94/commits/e6ddde929cbbeec92a8f32c342dee58ffb8f8f3e) and merged to `main`.

File: prefs.js [hunk 2]
```diff
@@ -46,6 +46,7 @@ function fillPreferencesWindow(window){
 
 	group = addGroup(page,'Appearance');
 	addSpinButton(settings,group,'max-string-length','Max string length (each field)',1,150,undefined);
+	addColorPicker(settings, group, 'font-color', 'Label Font color', "Specify alternative for color for label\nDefault falls back to color defined by user theme");
```
Comment (Moon-0xff) [inline prefs.js:49]: First line (`Specify alternative...`) of the tooltip could be dropped.

**Pull request merged**